EvidenceApi/V1/UseCase/CreateDocumentSubmissionUseCase.cs

```diff
@@ -20,48 +21,71 @@ public CreateDocumentSubmissionUseCase(IEvidenceGateway evidenceGateway, IDocume
             _documentsApiGateway = documentsApiGateway;
         }
 
-        public DocumentSubmissionResponse Execute(Guid evidenceRequestId, DocumentSubmissionRequest request)
+        public async Task<DocumentSubmissionResponse> ExecuteAsync(Guid evidenceRequestId, DocumentSubmissionRequest request)
         {
-            if (String.IsNullOrEmpty(request.DocumentType))
-            {
-                throw new BadRequestException("Document type is null or empty");
-            }
-
-            if (String.IsNullOrEmpty(request.ServiceName))
-            {
-                throw new BadRequestException("Service name is null or empty");
-            }
-
-            if (String.IsNullOrEmpty(request.RequesterEmail))
-            {
-                throw new BadRequestException("Requester email is null or empty");
-            }
+            ValidateRequest(request);
 
             var evidenceRequest = _evidenceGateway.FindEvidenceRequest(evidenceRequestId);
             if (evidenceRequest == null)
             {
                 throw new NotFoundException($"Cannot find evidence request with id: {evidenceRequestId}");
             }
 
+            var claimRequest = BuildClaimRequest(request);
+
+            var claim = await _documentsApiGateway.GetClaim(claimRequest).ConfigureAwait(true);
+
+            var documentSubmission = BuildDocumentSubmission(evidenceRequest, request, claim);
+
+            var createdS3UploadPolicy = await _documentsApiGateway.CreateUploadPolicy(claim.Document.Id).ConfigureAwait(true);
+            var createdDocumentSubmission = _evidenceGateway.CreateDocumentSubmission(documentSubmission);
+            createdDocumentSubmission.UploadPolicy = createdS3UploadPolicy;
+            return createdDocumentSubmission.ToResponse(request.DocumentType);
```

Comment:
Seeing this here, makes me think we might not need the upload policy to be on the domain model, similar to the way we do with the document types...

Maybe we just have it on the response, and it becomes an argument we pass to the response factory?
```suggestion
            return createdDocumentSubmission.ToResponse(request.DocumentType, createdS3UploadPolicy);
```